### PR TITLE
feat: Add tessoku book to contest table (#2776)

### LINKED ISSUE #2776: [Feat] 鉄則本の問題をインポートできるようにしましょう

### PR DESCRIPTION
**Pull request by KATO-Hiro**
close #2776
See #662

## Summary by CodeRabbit

* **New Features**
  * Added Tessoku Book contest support with provider registration, filtering, metadata, and custom display behavior.

* **Tests**
  * Added and refactored comprehensive tests for the Tessoku Book provider covering filtering, metadata, display, headers, round labeling, multi-source and edge cases, empty inputs, and parity with other providers; introduced centralized mock/test scaffolding.

* **Documentation**
  * Added a phased test plan detailing scope, rollout phases, test specs, and timeline.

### CODE REVIEW

File: src/test/lib/utils/test_cases/contest_table_provider.ts [hunk 1]
```diff
@@ -196,6 +196,40 @@ const [
...
+  tessoku_a62,
+  tessoku_a77,
+  tessoku_b07,
+  tessoku_b16,
+  tessoku_b28,
+  tessoku_b63,
+  tessoku_c09,
+  tessoku_c18,
+] = createContestTasks('tessoku-book', [
+  { taskId: 'tesesoku_book_a', taskTableIndex: 'A01', statusName: AC },
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply src/test/lib/utils/test_cases/contest_table_provider.ts]: ありがとう．修正します

File: docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+    { contest_id: 'tessoku-book', task_id: 'tesskoku_book_a', task_table_index: 'A01' },
+    { contest_id: 'tessoku-book', task_id: 'math_and_algorithm_ai', task_table_index: 'A06' },
+    { contest_id: 'tessoku-book', task_id: 'typical90_a', task_table_index: 'A77' },
+  ];
+  const filtered = provider.filter(mixedTasks);
+
+  // 検証: contest_id === 'tessoku-book' のみ
+  expect(filtered?.every((task) => task.contest_id === 'tessoku-book')).toBe(true);
+  expect(filtered).not.toContainEqual(expect.objectContaining({ contest_id: 'abc123' }));
+});
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md]: 修正します

File: docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+    { contest_id: 'tessoku-book', task_id: 'math_and_algorithm_ac', task_table_index: 'C09' },
+  ];
+  const table = provider.generateTable(tasks);
+
+  expect(table).toHaveProperty('tessoku-book');
+  expect(table['tessoku-book']).toHaveProperty('A06');
+  expect(table['tessoku-book']['A06']).toEqual(
+    expect.objectContaining({ problem_id: 'math_and_algorithm_ai' }),
+  );
+});
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md]: 修正します

File: docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md [hunk 4]
```diff
@@ -0,0 +1,763 @@
...
+    contest_id: 'tessoku-book',
+    problem_id: 'math_and_algorithm_ac',
+    problem_index: 'C09',
+  },
+  {
+    contest_id: 'tessoku-book',
+    problem_id: 'typical90_s',
+    problem_index: 'C18',
+  },
+];
```
[the remark this answers is not in the document]
Comment (KATO-Hiro) [reply docs/dev-notes/2025-11-01/add_and_refactoring_tests_for_contest_table_provider/plan.md]: 修正します

**Review by KATO-Hiro — COMMENTED** [review]
LGTM
